scalyr_agent/log_processing.py

```diff
@@ -1866,16 +1880,59 @@ def __apply_redaction_rule(self, line, redaction_rule):
         @return: A sequence of two elements, the line with the redaction applied (if any) and True or False
             indicating if a redaction was applied.
         """
+
+        def __hash_group(match):
+            print '&&&&&\n', match.groups()
+            if not match.groups():
+                print "NO GROUPS!!!!"
+                return scalyr_util.md5_digest(match.group() + redaction_rule.hash_salt)
+
+            # TODO NEED help here! for a replacement regular expression, how can we determine which
```

Comment:
@czerwingithub Need help here. Since i'm passing a callback into the `re.subn` i need to know which captured group should I be hashing and replacing the value for. Is there a good way to determine from the already existing configuration of `replacement` like `foo=\\1` to determine that is the group number 1 that needs to be captured and hashed? Let me know if the approach doesn't make sense.